src/jobs/jobs.controller.ts

```diff
@@ -567,13 +570,64 @@ export class JobsController {
   }
 
   /**
-   * Create job
+   * Transform a v4 job instance so that is compatible with v3
+   * @param job: a JobClass instance (v4)
+   * @returns a JobClassV3 instance
+   */
+  private mapJobClassV4toV3(job: JobClass): JobClassV3 {
+    const jobV3 = new JobClassV3();
+    // Map fields from v4 to v3
+    jobV3._id = job._id;
+    jobV3.id = job.id;
+    jobV3.emailJobInitiator = job.createdBy;
+    jobV3.type = job.type;
+    jobV3.creationTime = job.createdAt;
+    jobV3.jobStatusMessage = job.statusMessage;
+    jobV3.jobResultObject = job.jobResultObject;
+    jobV3.ownerGroup = job.ownerGroup;
+    // Remove datasetList from jobParams and assign it to jobV3.datasetList
+    const { datasetList, ...jobParams } = job.jobParams;
+    jobV3.datasetList = datasetList as DatasetListDto[];
+    // Store v4 configVersion in v3 jobParams
+    const newJobParams = { ...jobParams, configVersion: job.configVersion };
```

Comment:
v3 didn't have a config version. I suggest we omit it rather than squashing it into jobParams